Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/webapp/deploymentslot/DeploymentSlotNodePresenter.java

```diff
@@ -24,26 +24,52 @@
+import com.microsoft.azuretools.core.mvp.model.webapp.AzureWebAppMvpModel;
 import com.microsoft.azuretools.core.mvp.ui.base.MvpPresenter;
+import com.microsoft.tooling.msservices.serviceexplorer.azure.webapp.base.WebAppBaseState;
 
 public class DeploymentSlotNodePresenter<V extends DeploymentSlotNodeView> extends MvpPresenter<V> {
-    public void onStartDeploymentSlot(String subscriptionId, String webAppId, String slotName) throws IOException {
-        //todo
+    public void onStartDeploymentSlot(final String subscriptionId, final String webAppId,
+                                      final String slotName) throws IOException {
+        AzureWebAppMvpModel.getInstance().startDeploymentSlot(subscriptionId, webAppId, slotName);
+        final DeploymentSlotNodeView view = getMvpView();
+        if (!isViewDetached()) {
+            view.renderNode(WebAppBaseState.RUNNING);
+        }
     }
 
-    public void onStopDeploymentSlot(String subscriptionId, String webAppId, String slotName) throws IOException {
-        //todo
+    public void onStopDeploymentSlot(final String subscriptionId, final String webAppId,
+                                     final String slotName) throws IOException {
+        AzureWebAppMvpModel.getInstance().stopDeploymentSlot(subscriptionId, webAppId, slotName);
+        final DeploymentSlotNodeView view = getMvpView();
+        if (!isViewDetached()) {
+            view.renderNode(WebAppBaseState.STOPPED);
+        }
     }
 
-    public void onRestartDeploymentSlot(String subscriptionId, String webAppId, String slotName) throws IOException {
-        //todo
+    public void onRestartDeploymentSlot(final String subscriptionId, final String webAppId,
+                                        final String slotName) throws IOException {
+        AzureWebAppMvpModel.getInstance().restartDeploymentSlot(subscriptionId, webAppId, slotName);
+        final DeploymentSlotNodeView view = getMvpView();
+        if (!isViewDetached()) {
+            view.renderNode(WebAppBaseState.RUNNING);
+        }
     }
 
-    public void onRefreshNode() {
-        //todo
+    public void onRefreshNode(final String subscriptionId, final String webAppId,
+                              final String slotName) throws Exception {
+        final WebApp app = AzureWebAppMvpModel.getInstance().getWebAppById(subscriptionId, webAppId);
+        final DeploymentSlot slot = app.deploymentSlots().getByName(slotName);
+        final DeploymentSlotNodeView view = getMvpView();
+        if (!isViewDetached()) {
+            view.renderNode(WebAppBaseState.fromString(slot.state()));
+        }
     }
 
-    public void onSwapWithProduction() {
-        // todo
+    public void onSwapWithProduction(final String subscriptionId, final String webAppId,
+                                     final String slotName) throws IOException {
+        AzureWebAppMvpModel.getInstance().swapSlotWithProduction(subscriptionId, webAppId, slotName);
```

Comment:
How does the user know whether the operation is successful or not ?

Reply:
The notification "Swapping with production" keeps showing in the notification bar until the operation finished. Just like the Restart operation.

Reply:
" open the URL"  is aggressive to me. I think the notification is enough.